src/app/calendar-app/runbox-calendar-event.ts

```diff
@@ -69,137 +88,437 @@ export class RunboxCalendarEvent implements CalendarEvent {
+            recur.freq = frequency;
+            this.event.component.updatePropertyWithValue('rrule', recur);
+        } else {
             this.event.component.addPropertyWithValue('rrule', new ICAL.Recur({ freq: frequency }));
         }
     }
 
+    // How often does this repeat (every X freqs)
+    get recurInterval(): number {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        return recur ? recur.interval : 1;
+    }
+
+    set recurInterval(interval: number) {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        recur.interval = interval;
+        this.event.component.updatePropertyWithValue('rrule', recur);
+    }
+
+    // An UNTIL date, a COUNT or null = unset/repeats forever
+    get recurEnds(): Date | number {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        if (recur && recur.until) {
+            return recur.until.toJSDate();
+        }
+        if (recur && recur.count) {
+            return recur.count;
+        }
+        return null;
+    }
+
+    set recurEnds(end: Date | number) {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        recur.until = null;
+        recur.count = null;
+        if (typeof end === 'number' && end != null) {
+            recur.count = end;
+        }
+        if (end instanceof Date &&  end != null) {
+            // Must be a date (cant do typeof === 'Date' !?
+            const zone = this.event.startDate.zone;
+            const icaltime = ICAL.Time.fromJSDate(end);
+            icaltime.zone = zone;
+            recur.until = icaltime;
+        }
+        // else, everything null, repeats forever.
+        this.event.component.updatePropertyWithValue('rrule', recur);
+    }
+
+    // get "part"s, eg byday, bymonthday, bymonth, byyearday, byweekno
+
+    // One or more days of the week this rule could run on (or none)
+    // SU, MO etc (or can convert to days of week using .icalDayToNumericDay
+    // if a monthly BYDAY, could be "\+?1SU" or "-2TU" etc
+    get recursByDay(): string[] {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        if (!recur) {
+            return [];
+        }
+        const bydays = recur.getComponent('byday');
+        const rgx = /^([+-]?\d+)?(\w{2})$/;
+        const bydays_mapped = bydays.map((day) => {
+            const match = day.match(rgx);
+            const numth = match[1] ? match[1] : '0';
+            return { 'day': match[2], 'numth': numth };
+        });
+        return bydays_mapped;
+    }
+
+    // replace entire list of day(s)
+    set recursByDay(value: string[]) {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
+        recur.setComponent('byday', value);
+        this.event.component.updatePropertyWithValue('rrule', recur);
+    }
+
+    get recursByMonthDay(): string[] {
+        const recur = this.event.component.getFirstPropertyValue('rrule');
```

Comment:
Maybe these could get refactor into a private get/set `recursBy(what)` to avoid repetition?

Reply:
I thought of that but they're not all the same, I guess a few don't have special cases..

Reply:
Right, they are subtly distinct now that I've paid more attention to it.